### PR TITLE
Update ViewModels page, add Fody

### PR DESCRIPTION
**Pull request by worldbeater**
* Mentioned `PropertyChanged.Fody` and `ReactiveUI.Fody` packages
* Recommended always using `ReactiveObject` as a base class
* Updated code examples using new C# 7+ language features.

### CODE REVIEW

File: input/docs/handbook/view-models/index.md [hunk 1]
```diff
@@ -1,167 +1,112 @@
...
-Handling animation and Window position / minimize are also great examples of
-code that often should be in the View.
+So then, how do I decide what to put in the View? Concepts such as scroll position and control focus are great examples of code that is View-specific. Handling animation and Window position / minimize are also great examples of code that often should be in the View.
 
-Another common misconception is that of separation - while it is very
-important that the ViewModel have no reference to the View or any of the
-controls that the View creates, **the reverse is not true**. The View is free
-to be very tightly bound to the ViewModel, and in fact, it is often useful for
-the View to "reach into" the ViewModel via `WhenAny` and `WhenAnyObservable`.
+Another common misconception is that of separation - while it is very important that the ViewModel have no reference to the View or any of the controls that the View creates, **the reverse is not true**. The View is free to be very tightly bound to the ViewModel, and in fact, it is often useful for the View to "reach into" the ViewModel via `WhenAny` and `WhenAnyObservable`.
```
Comment (glennawatson) [inline input/docs/handbook/view-models/index.md]: WhenAnyValue is more common might be worth mentioning it.

File: input/docs/handbook/view-models/index.md [hunk 2]
```diff
@@ -1,167 +1,112 @@
...
+    get => name;
+    set => this.RaiseAndSetIfChanged(ref name, value);
 }
 ```
 
-Note, that unlike in other frameworks, they are **always written this way**,
-using the exact same boilerplate code. If you are attempting to put *anything*
-in the setter, you are almost certainly Doing It Wrong, and instead should be
-using `WhenAny` and `ToProperty` instead.
+Note, that unlike in other frameworks, they are **always written this way**, using the exact same boilerplate code. If you are attempting to put *anything* in the setter, you are almost certainly Doing It Wrong, and instead should be using `WhenAny` and `ToProperty` instead.
```
Comment (glennawatson) [inline input/docs/handbook/view-models/index.md]: Again might be worth mentioning WhenAnyValue etc.

File: input/docs/handbook/view-models/index.md [hunk 3]
```diff
@@ -1,167 +1,112 @@
...
-With the theory out of the way, let's see how to create ViewModels in
-ReactiveUI.
+With the theory out of the way, let's see how to create ViewModels in ReactiveUI.
 
-### Read-Write Properties
+# Creating a ViewModel class
 
-Properties that participate in change notification (i.e. that signal when they
-are changed), are written in the following way:
+**ReactiveObject** is the base object for ViewModel classes, and it implements `INotifyPropertyChanged`. In addition, ReactiveObject provides `Changing` and `Changed` Observables to monitor object changes. A typical ViewModel created using ReactiveUI framework should inherit from ReactiveObject:
```
Comment (RLittlesII) [inline input/docs/handbook/view-models/index.md]: Worth mentioning `IReactiveObject`.  Base classes are great, but we should try and code to abstractions.  Most of the methods that matter go against `IReactiveObject` for that reason.

File: input/docs/handbook/view-models/index.md [hunk 4]
```diff
@@ -1,167 +1,112 @@
...
 * "The error message should be cleared 10 seconds after it is displayed"
 * "The DirectMessageToSend object consists of the target user and the message
    to send""
 
-All of these statements are concise descriptions of parts of how your UI
-should work, and these statements can all be directly translated into Rx
-expressions in your ViewModel constructor.
+All of these statements are concise descriptions of parts of how your UI should work, and these statements can all be directly translated into Rx expressions in your ViewModel constructor.
+
+# Dealing with Boilerplate code
```
Comment (RLittlesII) [inline input/docs/handbook/view-models/index.md]: Change "Dealing with Boilerplate" to "Managing Boilerplate", sounds more inviting.

**Pull request merged**